R/derive_date_vars.R

```diff
@@ -1149,12 +1150,23 @@ compute_dtf <- function(dtc, dt) {
   valid_dtc <- is_valid_dtc(dtc)
   warn_if_invalid_dtc(dtc, valid_dtc)
 
+  # Find date portion
+  date_portion <- ifelse(grepl("T", dtc),
+    gsub("T", "", substr(dtc, 1, gregexpr("T", dtc))),
+    substr(dtc, 1, 10)
+  )
+  n_chr_date_portion <- nchar(date_portion)
+
+  # Location of the first instance of the double hyphen to determine if its month/day imputation
+  location_of_double_hyphen <- unlist(gregexpr("--", date_portion))
```

Comment:
This does not work for
```
> compute_dtf("2020----T00:00", ymd_hms("2020-05-15T00:00:00"))
[1] "M" NA 
```
Maybe we should use a different approach. Something like:
```
pdate <- get_partialdatetime(dtc)
impflags <- c(year = "Y", month = "M", day = "D")
impflags[[min(which(is.na(pdate)))]]
```

Reply:
Unfortunately, there are still cases which are not covered:
```
> compute_dtf("-----T00:00", ymd_hms("2020-05-15T00:00:00"))
[1] NA
> compute_dtf("--05--T00:00", ymd_hms("2020-05-15T00:00:00"))
[1] NA
```
I don't mind which way we go if it works for all cases and is readable.

Reply:
Are those acceptable formats? I'm pretty sure those are supposed to be caught by the 
`admiraldev:: is_valid_dtc()` and `admiraldev::warn_if_invalid_dtc()` calls. If that's the case, this becomes an `admiraldev` fix too

Reply:
It looks like `warn_if_invalid_dtc()` wants to parse out any of the missing year stuff, but `is_valid_dtc()` basically allows it? 
![image](https://github.com/pharmaverse/admiral/assets/43018762/866faffc-9e4d-4ba0-86ef-0031491d4661)
![image](https://github.com/pharmaverse/admiral/assets/43018762/38488679-e5fe-4be7-ab4f-fa44e0c5b6d5)

Reply:
In the beginning we ignored dates with missing elements in between (except dates like "2003---15"). The ideas was to release a first version quickly and handle the rare edge cases like "--12-15" later.
Cases like "--12-15" are explicitly mentioned in the ADaM IG. Thus I think we should cover them.

Reply:
I believe I've added this portion, at least with your reprex's above. I shall make an issue in `admiraldev` to reflect the removal of those cases on the "NOT handled ...." messaging